demux/demux_playlist.c

```diff
@@ -463,14 +497,62 @@ static bool scan_dir(struct pl_parser *p, char *path,
 
 static int parse_dir(struct pl_parser *p)
 {
-    if (!p->real_stream->is_directory)
-        return -1;
-    if (p->probing)
-        return 0;
+    int ret = -1;
+    struct stream *stream = p->real_stream;
+    int autocreate = AUTO_NONE;
+    p->pl->playlist_dir = NULL;
+    if (p->autocreate_playlist && p->real_stream->is_local_file && !p->real_stream->is_directory) {
+        bstr ext = bstr_get_ext(bstr0(p->real_stream->url));
+        switch (p->autocreate_playlist) {
+        case 1: // any
+            autocreate = AUTO_ANY;
+            break;
+        case 2: // filter
+            autocreate = AUTO_VIDEO | AUTO_AUDIO | AUTO_IMAGE;
```

Comment:
I thought this would match the value of `--directory-filter-types`. It's not supposed to?

Reply:
It could, yes.

Reply:
One thing, is that some users might want different filtering for autocreate and directory load

Reply:
Add yet another choice I guess? I don't have any name suggestions though. I personally expected `--autocreate-playlist=filter` to use `--directory-filter-types` and was surprised that it didn't.

Reply:
done